src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorShape.cs

```diff
@@ -435,7 +435,6 @@ public ReadOnlySpan<nint> Strides
         public nint AdjustToNextIndex(in TensorShape destinationShape, nint linearOffset, Span<nint> indexes)
         {
             Debug.Assert(indexes.Length >= Rank);
-            Debug.Assert(indexes.Length == destinationShape.Rank);
```

Comment:
When would this not be correct? We expect that `destinationShape` be either `this` or it be the shape we're "compatible with" and therefore broadcasting to.

Reply:
Its probably valid now. at one point during my itteration of changes it wasn't so I removed it. Let me re-add it.